### DISCUSSION TITLE
Lingui should evolve to have a 0 runtime footprint

### DISCUSSION THREAD

**Discussion by Bertg** [lingui/js-lingui#1283] [Ideas]
I wanted to discuss some ideas around Lingui, and what a "next" version could look like

What I love in the current version of Lingui, and that should be kept:
* PO files
* ICU message format
* Macro's

What could 4x look like?

* Become a bit opinionated 🙂 
* Lingui (in production) produces code with translations baked in, basically a package per language, with 0 runtime impact.
* Proper support for catalogs per component, right now only a single big catalog seems to work out of the box (let the translation tools deal with duplication)

**Comment by andrii-bodnar** [comment]
There is an issue that includes a tasks list that might be included in the v4 release - https://github.com/lingui/js-lingui/issues/1359

**Comment by Bertg** [comment]
~Closing in favour of the war in v4 and https://github.com/lingui/js-lingui/issues/1384~

This is not a ticket :)
